pkg/database/database.go

```diff
@@ -121,46 +141,55 @@ func (db *Database) QueryLogURLs(ctx context.Context, kind, apiVersion, namespac
 	uuid, err := strQueryPerformer.performSingleRowQuery(ctx, sb)
 
 	if errors.Is(err, sql.ErrNoRows) {
-		return []string{}, ResourceNotFoundError
+		return "", "", ResourceNotFoundError
 	}
 
 	if err != nil {
-		return nil, err
+		return "", "", err
 	}
 
-	podUuids, err := db.getOwnedPodsUuids(ctx, []string{uuid}, []string{})
-	if len(podUuids) == 0 {
-		return []string{}, ResourceNotFoundError
-	}
+	pods, err := db.getOwnedPodsUuids(ctx, []string{uuid}, []uuidKindDate{})
 	if err != nil {
-		return nil, err
+		return "", "", err
 	}
+	if len(pods) == 0 {
+		return "", "", ResourceNotFoundError
+	}
+
+	// Get the most recent pod from owned by the provided resource
+	sort.Slice(pods, func(i, j int) bool {
+		return pods[i].Date > pods[j].Date
+	})
 
 	sb = db.Selector.UrlSelector()
-	sb.Where(db.Filter.UuidsFilter(sb.Cond, podUuids))
-	return strQueryPerformer.performQuery(ctx, sb)
+	sb.Where(db.Filter.UuidsFilter(sb.Cond, []string{pods[0].Uuid}))
+	logUrls, err := logQueryPerformer.performQuery(ctx, sb)
+	if err != nil {
+		return "", "", err
+	}
+	if len(logUrls) >= 1 {
+		return logUrls[0].LogURL, logUrls[0].JsonPath, nil
+	} else {
+		return "", "", ResourceNotFoundError
+	}
```

Comment:
Why not return `nil` here rather than `""`?  It seems odd.

Reply:
because the returned type is a string. A string in golang can't be `nil`, its zero value is an empty string.

Reply:
I didn't know that. Thanks.